src/Lime.Transport.SignalR/SignalRTransportListener.cs

```diff
@@ -0,0 +1,224 @@
+            HttpProtocols httpProtocols,
+            SslProtocols sslProtocols,
+            Func<X509Certificate2, X509Chain, SslPolicyErrors, bool> clientCertificateValidationCallback,
+            TimeSpan? keepAliveInterval)
+        {
+            HttpConnectionDispatcherOptions httpConnectionDispatcherOptions = null;
+            HubOptions hubOptions = null;
+
+            return Host.CreateDefaultBuilder()
+                    .ConfigureWebHostDefaults(webBuilder =>
+                    {
+                        webBuilder.UseKestrel(serverOptions =>
+                        {
+                            foreach (var listenerUri in ListenerUris)
+                            {
+                                if (!IPAddress.TryParse(listenerUri.Host, out var ipAddress))
+                                {
+                                    ipAddress = IPAddress.Any;
+                                }
+
+                                var endPoint = new IPEndPoint(ipAddress, listenerUri.Port);
+                                serverOptions.Listen(endPoint, listenOptions =>
+                                {
+                                    listenOptions.Protocols = httpProtocols;
+
+                                    if (listenerUri.Scheme == URI_SCHEME_HTTPS)
+                                    {
+                                        listenOptions.UseHttps(tlsCertificate, httpsOptions =>
+                                        {
+                                            httpsOptions.SslProtocols = sslProtocols;
+                                            httpsOptions.ClientCertificateValidation = clientCertificateValidationCallback;
+                                        });
+                                    }
+                                });
+                            }
+
+                            serverOptions.AddServerHeader = false;
+                        })
+                        .SuppressStatusMessages(true)
+                        .ConfigureServices(services =>
+                        {
+                            services
+                            .AddLogging()
+                            .AddSingleton(sp => _transportChannel)
+                            .AddSingleton(sp => httpConnectionDispatcherOptions)
+                            .AddSingleton(sp => hubOptions)
+                            .AddSingleton(envelopeSerializer)
+                            .AddSingleton(new EnvelopeHubOptions { BackpressureLimit = _backpressureLimit })
+                            .AddSingleton(new ConcurrentDictionary<string, Channel<string>>())
+                            .AddSingleton<IUserIdProvider, RandomUserIdProvider>()
+                            .AddSignalR().AddHubOptions<EnvelopeHub>(options =>
+                            {
+                                hubOptions = options;
+                                options.KeepAliveInterval = keepAliveInterval;
+                            });
```

Comment:
Ident new lines 1 level
```suggestion
                            services
                                .AddLogging()
                                .AddSingleton(sp => _transportChannel)
                                .AddSingleton(sp => httpConnectionDispatcherOptions)
                                .AddSingleton(sp => hubOptions)
                                .AddSingleton(envelopeSerializer)
                                .AddSingleton(new EnvelopeHubOptions { BackpressureLimit = _backpressureLimit })
                                .AddSingleton(new ConcurrentDictionary<string, Channel<string>>())
                                .AddSingleton<IUserIdProvider, RandomUserIdProvider>()
                                .AddSignalR().AddHubOptions<EnvelopeHub>(options =>
                                {
                                    hubOptions = options;
                                    options.KeepAliveInterval = keepAliveInterval;
                                });
```